metaflow/plugins/aip/argo_utils.py

```diff
@@ -41,6 +63,52 @@ def run_argo_workflow(
     return argo_run_id, argo_run_uid
 
 
+def get_latest_workflow(
+    kubernetes_namespace: str,
+    project_name: Optional[str] = None,
+    branch_name: Optional[str] = None,
+    template_name_prefix: Optional[str] = None,
+):
+    # TODO:
+    # - Add filter by project_id instead of project name - project_id is not added as a label yet.
+    # - Add filter by flow_name - flow_name is not added as a label yet.
+    client = ArgoClient(namespace=kubernetes_namespace)
+
+    templates = client.list_workflow_template()["items"]
+    templates = [
+        template
+        for template in templates
+        if (
+            not project_name
+            or template["metadata"]["labels"]["gitlab.zgtools.net/project-name"]
+            == project_name
+        )
+        and (
+            not branch_name
+            or template["metadata"]["labels"]["gitlab.zgtools.net/branch-name"]
+            == branch_name
+        )
+        and (
+            not template_name_prefix
+            or template["metadata"]["name"].startswith(template_name_prefix)
+        )
```

Comment:
The logic hear looks clean and easy to follow.

One question: Is it possible that a key like "gitlab.zgtools.net/branch-name" won't exist in a template? If it is possible, can you fix by changing line 88 for example to `or template["metadata"]["labels"].get("gitlab.zgtools.net/branch-name")`?

Reply:
Thanks for catching this. Fixed!